mlcube/mlcube/main.py

```diff
@@ -1,42 +1,76 @@
-import sys
-from time import sleep
-from pathlib import Path
-import pprint
-import logging
-
+import os
 import click
-from halo import Halo
+import logging
 import coloredlogs
-
+from halo import Halo
+from pathlib import Path
+from typing import Optional
 from mlcube.check import check_root_dir
+from mlcube.common.mlcube_metadata import (MLCubeFS, CompactMLCube)
+
 
-@click.group(name='mlcube')
-@click.option('--log-level', default="DEBUG", help="Log level for the app.")
+logger = logging.getLogger(__name__)
+
+
+@click.group(name='mlcube', help="MLCube 📦 is a packaging tool for ML models")
+@click.option('--log-level', default="INFO", help="Log level for the app.")
 def cli(log_level: str):
-    """
-    MLCube 📦 is a packaging tool for ML models
-    """
-    logger = logging.getLogger(__name__)
     click.echo(f"Log level is set to - {log_level}")
     coloredlogs.install(level=log_level)
 
-@cli.command()
-@click.argument('path_to_mlcube')
+
+@cli.command(name='verify', help='Verify MLCube metadata.')
+@click.option('--mlcube', required=False, type=str, help='MLCube path.')
 @Halo(text="", spinner="dots")
-def verify(path_to_mlcube: str):
-    """
-    Verify MLCube metadata
-    """
+def verify(mlcube: Optional[str]) -> None:
     logging.info("Starting mlcube metadata verification")
-    metadata, verify_err = check_root_dir(
-    Path(path_to_mlcube).resolve().as_posix())
-
+    mlcube_path = CompactMLCube(mlcube).unpack().mlcube_fs.root
+    metadata, verify_err = check_root_dir(mlcube_path)
     if verify_err:
         logging.error(f"Error verifying mlcube metadata: {verify_err}")
         logging.error(f"mlcube verification - FAILED!")
         raise click.Abort()
-
     logging.info('OK - VERIFIED')
 
+
+@cli.command(name='pull', help='Pull MLCube from some remote location.')
+@click.option('--mlcube', required=True, type=str, help='Location of a remote MLCube.')
```

Comment:
I'm not too familiar with click - but it looks really helpful. Thanks for introducing me to it.